### PR TITLE
Add TuxedoJS Overview doc.

### PR DESCRIPTION
**Pull request by drabinowitz**

### CODE REVIEW

File: docs/Overview.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+# TuxedoJS In a Nutshell
+>TuxedoJS is a client-side CommonJS based web development framework built on Facebook's awesome `React` view-layer and `Flux` Architecture. It provides a semantic interface for working with `Flux` and augmented `React` components for managing different aspects of the view logic.  Tux also provides some powerful prebuilt animation components that make animating transitions in `React`  a synch. Lastly, TuxedoJS leverages CommonJS modules to provide a completely modular framework which means, unlike other popular frameworks, you only load the pieces of Tux you actually intend to use.
```
Comment (Cheerazar) [inline docs/Overview.md]: Change the 2 spaces after the period to 1. Apply throughout the document.

Comment (Cheerazar) [inline docs/Overview.md]: Remove extra space after '...animating transitions in `React`'

File: docs/Overview.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+## The React View Layer
+TuxedoJS leverages `React` for its views.  `React` is Facebook's new high performance view library that provides really clean and semantic language for defining how your views should be rendered and when/how they should update. You can read more about `React` [here](http://facebook.github.io/react/).  `React` is a purely Javascript based view layer.  It uses a virtual DOM to compare the current DOM state to the new DOM state when updating so that it can submit only the minimal changes required to update the DOM. Lastly, `React` is often written using [JSX](http://facebook.github.io/react/docs/displaying-data.html) syntax, which allows you to write pseudo-html in Javascript.
```
Comment (Cheerazar) [inline docs/Overview.md]: Maybe mention that changing the actual DOM is expensive and the virtual DOM allows for performance increases.

File: docs/Overview.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+```
```
Comment (Cheerazar) [inline docs/Overview.md]: Add javascript as the language for the code block.

File: docs/Overview.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+The Dispatcher manages all the view and server events (in `Flux` these are called `actions`) and then dispatches those actions.  Every store in a `Flux` application listens to the Dispatcher and simply decides for itself which actions are relevant to it.  Then, the stores that have updated emit update events to inform their views that they need to come and get the new data. This design dramatically simplifies data flow in an app and allows stores to fully encapsulate their data with as little coupling as possible.  Normally, in a `Flux` application the Views will respond to user interaction by triggering `Action Helpers`, which usually will make Ajax requests or some other kind of asynchronous request and then dispatch the corresponding `Action` when the request completes.
```
Comment (Cheerazar) [inline docs/Overview.md]: Recommend mentioning that the stores register themselves with the dispatcher.

File: docs/Overview.md (hunk 2 above)
Comment (Cheerazar) [inline docs/Overview.md]: Change all references of Javascript to JavaScript.

File: docs/Overview.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+TuxedoJS solves these problems by wrapping `React` and `Flux` together into a single framework that abstracts away all of the boilerplate and complexity associated with getting a `Flux` application up and running and communicating with React.  Additionally, Tux Provides powerful augmented `React` Classes that are designed to handle different aspects of the app frontend such as managing state and listening to events from stores.  The TuxedoJS app architecture looks like the following:
+
+NEED IMAGE
+
+Let's dive in to what these different pieces actually represent.
+
+### The Modular Framework Structure
+In the below examples we are using the `require` syntax associated with `CommonJS` modules.  Of note is the fact that we are never writing the line `require('tux');`. That's because there is no global object to require for TuxedoJS.  Instead, Tux is broken up into individual modules that handle a particular aspect of your application.  What's the advantage of this?
+
+Firstly, it allows you to write semantic require statements that make it extremly clear what tux modules you need for your individual app components.
```
Comment (Cheerazar) [inline docs/Overview.md]: Correct spelling of extremely.

File: docs/Overview.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+      render: function () {
+        return (
+          <h1>Hello World!</h1>
+        );
+      }
+    });
+```
+
+## The Flux Architecture
+TuxedoJS is built on Facebook's [Flux](https://facebook.github.io/flux/) architecture.  The two key concepts of `Flux` are one-way data flow and the `Dispatcher`.  With one way data flow, views and `stores` (in `Flux` models/collections are replaced with stores) do not communicate back and forth as in standard MVC applications.  Instead, views communicate with stores through a Dispatcher.  This Dispatcher can be thought of as a Controller in normal MVC applications. However, unlike in standard MVC architectures, there is only one Dispatcher.
```
Comment (Cheerazar) [inline docs/Overview.md]: Should you put references to TuxedoJS into \` as both React and Flux are getting that treatment.

Comment (Cheerazar) [inline docs/Overview.md]: Check to see if all references of React and Flux have been wrapped in \`.

File: docs/Overview.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+TuxedoJS solves these problems by wrapping `React` and `Flux` together into a single framework that abstracts away all of the boilerplate and complexity associated with getting a `Flux` application up and running and communicating with React.  Additionally, Tux Provides powerful augmented `React` Classes that are designed to handle different aspects of the app frontend such as managing state and listening to events from stores.  The TuxedoJS app architecture looks like the following:
+
+NEED IMAGE
```
Comment (Cheerazar) [inline docs/Overview.md]: Remove if we dont' get an image in time.

File: docs/Overview.md (hunk 5 above)
Comment (Cheerazar) [inline docs/Overview.md]: Should tux and and Tux be wrapped in \`?

File: docs/Overview.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+```
+    var Actions = require('tux/Actions');
+    var Stores = require('tux/Stores');
+    var React = require('tux/React');
+    var Architecture = require('tux/Architecture');
+    var Fly = require('tux/Animations/Fly');
+    var FadeDownBig = require('tux/Animations/Fade/DownBig');
+```
+
+### TuxedoJS provides a novel approach to store dependencies that allows you to write out all of your store relationships in one file using a simple and declarative syntax.
```
Comment (Cheerazar) [inline docs/Overview.md]: This is an extremely long heading. Recommend slimming down a little bit.

File: docs/Overview.md [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+This turns the process of managing store relationships from frustrating and difficult to easy and self-documenting, allowing you and future engineers in your codebase to easily reason about your application architecture. TuxedoJS is the first framework we know of that allows you to write out your model dependencies in such a clean and simple syntax and we are extremely excited to use this syntax in building complex web apps.
```
Comment (Cheerazar) [inline docs/Overview.md]: Recommend changing '...future engineers in your...' to '...future engineers using your codebase...'

File: docs/Overview.md (hunk 8 above)
Comment (Cheerazar) [inline docs/Overview.md]: Recommend mentioning that these are inter-store relationships.

File: docs/Overview.md [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+```
+    var Actions = require('tux/Actions');
+
+    var roomActions = Actions.createActionCategory({
+      category: 'rooms',
+      source: 'room views',
+      actions: ['get', 'add', 'remove']
+    });
+
+    //the dispatchGet input is the method that will actually dispatch our rooms/get action. Thus, here we are submitting an ajax request and then dispatching the results of that request for our rooms/get action whenever roomActions.get method is invoked
```
Comment (Cheerazar) [inline docs/Overview.md]: Change any references of ajax to Ajax.

File: docs/Overview.md [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+      get: roomStore.onGetDispatch
+    });
+
+    module.exports = roomStore;
+```
+
+### TuxedoJS provides powerful augmented `React` syntax which makes it extremely easy to connect your state managing views with your stores
+TuxedoJS provides opinionated `React` classes with built in convenience methods for performing distinct operations in your app. The augmented class in the example below possesses tools for easily and semantically connecting with Tux Stores.
+
+Note here that we are storing the result of `require('tux/React')` in a variable called `React` here.  This is to take advantage of React's beautiful JSX syntax, which automatically converts JSX to `React.METHOD NAME`.
```
Comment (Cheerazar) [inline docs/Overview.md]: Recommend changing Note to NOTE and highlighting that worde with \*\*.

File: docs/Overview.md [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+
+      render: function () {
+        return (
+          <Rooms rooms={this.state.rooms} />
+        );
+      }
+    });
+
+    module.exports = RoomViewOwner;
+```
```
Comment (Cheerazar) [inline docs/Overview.md]: Do you want to go into detail about what the `OwnerClass` provides as well as the `MutableClass`?

File: docs/Overview.md [hunk 13]
```diff
@@ -0,0 +1,213 @@
...
+
+`Tux` even provides a class designed to take advantage of the performance boosting tools in `React` for performance critical components in your application.
+
+```
```
Comment (Cheerazar) [inline docs/Overview.md]: Add javascript markdown tag.

File: docs/Overview.md [hunk 14]
```diff
@@ -0,0 +1,213 @@
...
+            <h2>{this.props.room.name}</h2>
+            <button onClick={this.handleDelete}>Delete</button>
+          </div>
+        );
+      }
+    });
+```
+
+## Next Steps
+Hopefully now you are a little excited about `TuxedoJS` and how you can use it in your apps. From here you can head to [getting started](./getting-started) to build a basic Todo app in `tux`. Alternatively, take a look at our deployed [Tux Chat App](TuxedoJS.github.io/TuxChatApp) and [Tux Todo App](TuxedoJS.github.io/TuxTodoApp). Lastly, check out our [api documentation](./docs) to get a deep dive with `Tux`.
```
Comment (Cheerazar) [inline docs/Overview.md]: Recommend changing lower case \`tux\` to uppercase \`Tux\` as it is used as the last word.

File: docs/Overview.md [hunk 15]
```diff
@@ -0,0 +1,213 @@
...
+
+`TuxedoJS` solves these problems by wrapping `React` and `Flux` together into a single framework that abstracts away all of the boilerplate and complexity associated with getting a `Flux` application up and running and communicating with React. Additionally, `Tux` Provides powerful augmented `React` Classes that are designed to handle different aspects of the app frontend such as managing state and listening to events from stores. The `TuxedoJS` app architecture looks like the following:
+
+NEED IMAGE
```
Comment (Cheerazar) [inline docs/Overview.md]: If you don't think we're going to get this in there right now, remove it and add a chore for it.

File: docs/Overview.md [hunk 16]
```diff
@@ -0,0 +1,213 @@
...
+`TuxedoJS` provides opinionated `React` classes with built in convenience methods for performing distinct operations in your app. The augmented class in the example below possesses tools for easily and semantically connecting with `Tux` Stores.
+
+**NOTE** here that we are storing the result of `require('tux/React')` in a variable called `React` here. This is to take advantage of React's beautiful JSX syntax, which automatically converts JSX to `React.METHOD NAME`.
+
+```javascript
+    var React = require('tux/React');
+    var roomStore = require('../stores/roomStore');
+    var roomActions = require('../actions/roomActions');
+
+     //for this example let's imagine we have created a Rooms and RoomCreateForm TuxReact class
```
Comment (Cheerazar) [inline docs/Overview.md]: Out dent one space.

File: docs/Overview.md [hunk 17]
```diff
@@ -0,0 +1,213 @@
...
+          </div>
+        );
+      }
+    });
+```
+
+## Next Steps
+Hopefully now you are a little excited about `TuxedoJS` and how you can use it in your apps. From here you can head to [getting started](./getting-started) to build a basic Todo app in `tux`. Alternatively, take a look at our deployed [Tux Chat App](TuxedoJS.github.io/TuxChatApp) and [Tux Todo App](TuxedoJS.github.io/TuxTodoApp). Lastly, check out our [api documentation](./docs) to get a deep dive with `Tux`.
+
+
```
Comment (Cheerazar) [inline docs/Overview.md]: Remove extra white space.
